### PR TITLE
FAB-1354: rework agents list-activations command to allow for filtering options by state or time

### PR DESCRIPTION
**Pull request by clee-cs**
As of now, only the `--json` option works.. need to update the processing of the response for the pretty table (guess it wasn't updated after we decided to only list activationId's)

Here's the current output from a project in dci-dev:
```
clee@CS-0238 20:32:40 ~/projects/cortex5/cortex-cli  (FAB-1354)
$ cortex agents list-activations --project run-sh-dci daemon-test
┌──────────────────────────────────────┬────────────────────┬───────────────┬───────────────┬───────────────┬──────────────────────────────────────┐
│ Activation Id                        │ Type               │ Status        │ Start         │ End           │ Session Id                           │
├──────────────────────────────────────┼────────────────────┼───────────────┼───────────────┼───────────────┼──────────────────────────────────────┤
│ -                                    │ -                  │ -             │ -             │ -             │ -                                    │
├──────────────────────────────────────┼────────────────────┼───────────────┼───────────────┼───────────────┼──────────────────────────────────────┤
│ -                                    │ -                  │ -             │ -             │ -             │ -                                    │
└──────────────────────────────────────┴────────────────────┴───────────────┴───────────────┴───────────────┴──────────────────────────────────────┘
(base) 
clee@CS-0238 20:32:58 ~/projects/cortex5/cortex-cli  (FAB-1354)
$ cortex agents list-activations --project run-sh-dci daemon-test --json
[
  "7c0c48ef-ab62-469d-8866-18277278f2be",
  "56cc82d0-3f60-408e-827e-bf4da476e6ab"
]
(base) 
clee@CS-0238 20:33:06 ~/projects/cortex5/cortex-cli  (FAB-1354)
$ cortex agents list-activations --project run-sh-dci daemon-test --json --startBefore 1616661100815
[
  "7c0c48ef-ab62-469d-8866-18277278f2be"
]
(base) 
clee@CS-0238 20:38:01 ~/projects/cortex5/cortex-cli  (FAB-1354)
$ cortex agents list-activations --project run-sh-dci daemon-test --json --startBefore 1616661100815 --status complete
[]
(base) 
clee@CS-0238 20:38:18 ~/projects/cortex5/cortex-cli  (FAB-1354)
$ cortex agents list-activations --project run-sh-dci daemon-test --json --startBefore 1616661100815 --status error
[
  "7c0c48ef-ab62-469d-8866-18277278f2be"
]

```

# Checklist:
Please check you fulfill ALL of the relevant checkboxes
- [ ] Notified docs of any potential USER-facing changes
- [ ] Added short description of the change - with relevant motivation and context. 
- [ ] Branch has the ticket number in its name (along with a ticket summary)
- [ ] Commented the code, particularly in hard-to-understand areas
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Ran `npm test` and it passes
- [ ] Changes generate no new warnings

### CODE REVIEW

**Review by mcriscolo-cs — APPROVED** [review]

**Review by daldridge-cs — APPROVED** [review]

**Review by jlara-cs — APPROVED** [review]
lgtm

File: bin/cortex-agents.js [hunk 1]
```diff
@@ -128,18 +128,24 @@ program
...
+    .command('list-activations <agentName>')
+    .description('List activations for an agent')
     .option('--no-compat', 'Ignore API compatibility checks')
     .option('--color [on/off]', 'Turn on/off colors for JSON output.', 'on')
     .option('--profile [profile]', 'The profile to use')
     .option('--project [project]', 'The project to use')
     .option('--json', 'Output results using JSON')
     .option('--query [query]', 'A JMESPath query to use in filtering the response data.')
-    .action(withCompatibilityCheck((snapshotId, options) => {
+    .option('--startBefore [timestamp]', 'A timestamp that activations must have started before.')
```
Comment (smichalski-cs) [inline bin/cortex-agents.js:141]: Include the word filter in the descriptions: "filters activations to include those that started before the specified timestamp."
"filters activations to include those that started after the specified timestamp"
"filters activations to include those that ended before the specified timestamp"
"filters activations to include those that ended after the specified timestamp"

File: bin/cortex-agents.js [hunk 2]
```diff
@@ -128,18 +128,24 @@ program
...
+    .option('--startAfter [timestamp]', 'A timestamp that activations must have started after.')
+    .option('--endBefore [timestamp]', 'A timestamp that activations must have ended before.')
+    .option('--endAfter [timestamp]', 'A timestamp that activations must have ended after.')
+    .option('--status [status]', 'Filter activations by status [complete|failed].')
```
Comment (smichalski-cs) [inline bin/cortex-agents.js:145]: filters (add an s)

**Review by smichalski-cs — APPROVED** [review]
some minor wording changes

**Review by daldridge-cs — APPROVED** [review]

**Pull request merged**